### PR TITLE
Unit tests for `LocalPravegaEmulator`

### LINKED ISSUE #1881: Add test for standalone pravega
**Problem description**
We need a simple test to verify that standalone pravega comes up. Otherwise, we can have changes breaking it and going unnoticed, which is problematic.

**Problem location**
Pravega standalone.

**Suggestions for an improvement**
Add a test case that, for example, checks via the REST API that the controller comes up. It would be even better if we could check that we can create streams, write and such.

### PR DESCRIPTION
**Pull request by arvindkandhare**
**Change log description**
Creates a `LocalPravegaEmulator` deployment 
Creates a sample scope and stream

**Purpose of the change**
To test Pravega standalone deployment. This fixes #1881 

**What the code does**
Adds unit test for LocalPravegaEmulator.

**How to verify it**
The unit tests pass.

### CODE REVIEW

File: standalone/src/main/java/io/pravega/local/LocalPravegaEmulator.java [hunk 1]
```diff
@@ -92,8 +94,9 @@ public void close() throws Exception {
 
     /**
      * Start controller and host.
+     * @throws Exception passes on the exception thrown by `inProcPravegaCluster`
      */
-    private void start() throws Exception {
+    public void start() throws Exception {
```
Comment (andreipaduroiu) [inline standalone/src/main/java/io/pravega/local/LocalPravegaEmulator.java]: 1. Add `@VisibleForTesting` since the only reason this is non-private is because of unit tests.
2. Make package-private, not public (i.e., no visibility modifier). That should suffice.

File: standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Unit tests for standalone
+ */
+@Slf4j
+public class InProcPravegaClusterTest {
+    LocalPravegaEmulator localPravega;
```
Comment (andreipaduroiu) [inline standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java]: private

File: standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    @Before
+    public void setUp() throws Exception {
+        ServiceBuilderConfig config = ServiceBuilderConfig
+                .builder()
+                .include(System.getProperties())
```
Comment (andreipaduroiu) [inline standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java]: This is a unit test to which we shouldn't pass any sys props. Why is this necessary?

File: standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                                   .scope(scope)
+                                   .streamName(streamName)
+                                   .scalingPolicy(ScalingPolicy.fixed(numSegments))
+                                   .build());
+        log.info("Created stream: " + streamName);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        localPravega.close();
```
Comment (andreipaduroiu) [inline standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java]: verify if this is null. The `@Start` method could have failed, and also the `localPravega` is not final, so any weird code could have reset it to null.

File: standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        String streamName = "Stream";
+        int numSegments = 10;
+
+        @Cleanup
+        StreamManager streamManager = StreamManager.create(URI.create(
+                localPravega.getInProcPravegaCluster().getControllerURI()
+        ));
+
+        streamManager.createScope(scope);
+        streamManager.createStream(scope, streamName,
```
Comment (andreipaduroiu) [inline standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java]: `createStream` returns a boolean. Verify the value is `true`

Comment (arvindkandhare) [reply standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java]: Fixed. Added an `assertTrue`.

File: standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java (hunk 4 above)
Comment (arvindkandhare) [reply standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java]: Fixed.

File: standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java (hunk 3 above)
Comment (arvindkandhare) [reply standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java]: This is to ensure that the test is as near to `standalone` as possible.

File: standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java (hunk 2 above)
Comment (arvindkandhare) [reply standalone/src/test/java/io/pravega/local/InProcPravegaClusterTest.java]: Fixed.

File: standalone/src/main/java/io/pravega/local/LocalPravegaEmulator.java (hunk 1 above)
Comment (arvindkandhare) [reply standalone/src/main/java/io/pravega/local/LocalPravegaEmulator.java]: Fixed. Made it packave-private.
